### ISSUE TITLE
Even when the computer is in sleep mode, SSO sessions can expire and result in multiple session requests being opened in the browser

### ISSUE THREAD

**Issue by yasithab** [synfinatic/aws-sso-cli#644]
**Output of `aws-sso version`:**
```
AWS SSO CLI Version 1.14.2 -- Copyright 2021-2023 Aaron Turner
792f11ec86a5ce6683ba2df7cb544e4f76f2431b (v1.14.2) built at 2023-10-20T04:29:15+0000
```

**Describe the bug:**
Even when the computer is in sleep mode, SSO sessions can expire and result in multiple session requests being opened in the browser. For instance, if the computer is asleep for 12 hours, there may be more than 10 session requests visible in the browser (refer to the attached screenshot). How can I avoid this? Is there a command to open the browser only when I specifically want to refresh the token?

**To Reproduce:**
Computer is in sleep mode and SSO session expired.

**Expected behavior:**
The desired behavior is to have the browser open only when the computer wakes up
OR 
When a specific command is executed to refresh the token. For instance, in the morning before starting work, running a command will open the browser and allow for token refreshing.

**Screenshots:**
<img width="1728" alt="1" src="https://github.com/synfinatic/aws-sso-cli/assets/9078175/ba91092e-323c-4dfb-b68a-4e0e2e1ce33c">

**Desktop**
 - OS: macOS
 - Version 14.0 (23A344)

**Additional context:**
N/A

**Contents of `~/.aws-sso/config.yaml`:**
```
SSOConfig:
    PF:
        SSORegion: us-east-1
        StartUrl: https://d-0123456789.awsapps.com/start
        DefaultRegion: ap-southeast-1
DefaultRegion: ap-southeast-1
ConsoleDuration: 720
CacheRefresh: 168
AutoConfigCheck: true
UrlAction: granted-containers
ConfigProfilesUrlAction: granted-containers
UrlExecCommand:
    - /Applications/Firefox.app/Contents/MacOS/firefox
    - "%s"
LogLevel: error
HistoryLimit: 10
HistoryMinutes: 1440
ProfileFormat: "{{ .AccountIdPad }}:{{ .RoleName }}"
FullTextSearch: true
```

**Comment by synfinatic** [comment]
So sounds like you have one or more processes trying to use IAM credentials and you're using `$AWS_PROFILE` to use them?

Options are:
1. Don't run processes when you're about to put the computer to sleep.
2. Use some other option other than`$AWS_PROFILE` which invokes `aws-sso process`
3. Wait until aws-sso v2 is releases which won't do this

**Comment by yasithab** [comment]
@synfinatic I appreciate your prompt response. Although I don't have any specific processes running, I have set up aliases in my ~/.zshrc file which use the `aws --profile` flag. Typically, I keep the terminal application open without closing it.

Also, I have kube configs with profile flag

```
aws --profile 01234567890:AdministratorAccess eks update-kubeconfig --region ap-southeast-1 --name kube-cluster --alias development
```

Is it possible that this is the issue? 

Anyway, Yes I can wait till V2. 

**Thank you very much for your time on this project :)**

**Comment by synfinatic** [comment]
Well, I'm just guessing on the issue.  All I can say is that something is calling `aws-sso` and so it is triggering the auto-login feature.   I can't tell you what exactly is doing that on your system for obvious reasons.   

As for is `aws --profile 01234567890:AdministratorAccess eks ...` causing it... well maybe?  Would have to look at the contents of you `~/.aws/config` file.

**Comment by yasithab-pf** [comment]
@synfinatic Really sorry for the late reply. Please find the content of `~/.aws/config` below.

<img width="1041" alt="Screenshot 2023-10-29 at 3 58 28 PM" src="https://github.com/synfinatic/aws-sso-cli/assets/111726818/b376a1e7-8a61-4d3d-9c88-223caf7924f4">

**Comment by yasithab-pf** [comment]
@synfinatic  Really sorry for the late reply. Please find the content of `~/.aws/config` below.

<img width="1041" alt="Screenshot 2023-10-29 at 3 58 28 PM" src="https://github.com/synfinatic/aws-sso-cli/assets/111726818/e03cf7ee-a101-4bd6-a94f-fbdc021395a5">

**Comment by synfinatic** [comment]
So does `01234567890:AdministratorAcces` from your `aws` command match one of the profiles in your `~/.aws/config`?  If so, then yes, that would cause it since as you can see that `$AWS_PROFILE` is calling `aws-sso` to fetch the credentials.  If they are expired, then it will try to refresh them which causes the browser to open a tab.

**Comment by yasithab** [comment]
@synfinatic I am closing this issue as it has been resolved in version v2.0.0-beta1. Thank you for your time and effort on this project :)
